packages/xstate-react/src/useActor.ts

```diff
@@ -1,69 +1,44 @@
-import { useState, useEffect, useMemo, useRef, useCallback } from 'react';
-import { ActorRef, Sender, ActorRefLike } from './types';
-import { EventObject, Interpreter, Actor } from 'xstate';
-import { fromService } from './useService';
-
-function resolveActor(
-  actorLike:
-    | ActorRef<any, any>
-    | ActorRefLike<any, any>
-    | Interpreter<any, any>
-    | Actor
-): ActorRef<any, any> & { deferred?: boolean } {
-  if (actorLike instanceof Interpreter) {
-    return fromService(actorLike);
-  }
-
-  if (!('current' in actorLike)) {
-    return {
-      stop: () => {
-        /* do nothing */
-      },
-      ...actorLike,
-      subscribe: actorLike.subscribe as any, // TODO: fix (subscribe will work, just improperly typed)
-      name: actorLike.id!,
-      current: actorLike.state!
-    };
-  }
-
-  return actorLike;
-}
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { Sender, ActorRefLike } from './types';
+import { EventObject, Actor } from 'xstate';
 
 export function useActor<TEvent extends EventObject, TEmitted = any>(
-  actorLike: ActorRefLike<TEvent, TEmitted> | Actor
+  actorRef: ActorRefLike<TEvent, TEmitted> | Actor,
+  getSnapshot: (actor: typeof actorRef) => TEmitted = (a) =>
+    'state' in a ? a.state : (undefined as any)
 ): [TEmitted, Sender<TEvent>] {
-  const actor = useMemo(() => resolveActor(actorLike), [actorLike]);
-  const deferredEventsRef = useRef<EventObject[]>([]);
-  const [current, setCurrent] = useState(actor.current);
+  // const actor = useMemo(() => resolveActor(actorLike), [actorLike]);
+  const deferredEventsRef = useRef<TEvent[]>([]);
+  const [current, setCurrent] = useState(() => getSnapshot(actorRef));
```

Comment:
in theory, we allow for changing `actorRef` between renders but this `current` won't sync to it so one would see stale values rendered until a new actor emits a value

Reply:
Good call.